src/Compilers/Core/Portable/Text/StringTextWriter.cs

```diff
@@ -33,9 +33,13 @@ public override Encoding Encoding
             get { return _encoding!; }
         }
 
-        public override SourceText ToSourceText()
+        public override SourceText ToSourceTextAndClear()
         {
-            return new StringText(_builder.ToString(), _encoding, checksumAlgorithm: _checksumAlgorithm);
+            var sourceText = new StringBuilderText(_builder, _encoding, checksumAlgorithm: _checksumAlgorithm);
+
+            _builder = new StringBuilder();
```

Comment:
do we actually need to reset this?  do we ever have any StringTextWRiters that are usable after this point?  We could just that the type is invalid after ToSourceText.

Reply:
I don't *need* to reset it, but it removes the concern of callers invalidly using this type and creating SourceText objects that change underneath them.

Reply:
Sorry.  Let me clarify.  I think it's *good* that it is swapped out.  However, i'm suggesting it be swapped out with 'null', not a new string-builder that will be garbage.  Effectively, makign this type a one-and-done value.

Reply:
Sure, I could make _builder nullable and set it to null here instead of creating a new StringBuilder. I guess I was being lazy as that will involve checking _builder at all usage points, but I can do that if the consensus indicates that is preferred.

Reply:
Alternatively, you say it's non-null, and allow it to blow up once in the 'done/disposed' state.

Reply:
These changes went away upon benchmarking analysis.